### ISSUE TITLE
[docdb] Add TSAN thread annotations to YQLPartitionsVTable

### ISSUE THREAD

**Issue by bmatican** [yugabyte/yugabyte-db#7110]
As a follow up to #7109 we realized we probably should leverage thread annotations to make sure we don't break this in the future. cc @nspiegelberg

**Comment by bozhiyou** [comment]
Hey, can I take this as my first issue?

**Comment by bmatican** [comment]
@bozhiyou definitely! you can see examples in `src/yb/master/catalog_manager.h`, with `GUARDED_BY(lock_);` and `REQUIRES_SHARED(lock_);`

do note, for this, you would need to be able to build the code in TSAN build type.
